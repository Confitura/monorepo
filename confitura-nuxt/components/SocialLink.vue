<template>
  <a
    :href="url"
    v-if="id"
    target="_blank"
    rel="noopener"
    class="link"
    :class="`link--${theme}`"
  >
    <i :class="icon"></i>
  </a>
</template>

<script setup lang="ts">

const { type = 'twitter', id, theme = 'black' } = defineProps<{ type: string, id: string, theme?: string }>()

const social = {
  twitter: {
    icon: 'fab fa-twitter-square',
    url: 'https://twitter.com/'
  },
  facebook: {
    icon: 'fab fa-facebook-square',
    url: 'https://www.facebook.com/'
  },
  github: {
    icon: 'fab fa-github-square',
    url: 'https://www.github.com/'
  },
  www: {
    icon: 'fas fa-globe',
    url: ''
  }
}

let icon = computed(() => social[type].icon)
let url = computed(() => social[type].url + id)
</script>
<style scoped lang="scss">
@use "~/assets/colors" as *;

.link {
  &#{&}--black {
    color: #000000;
  }

  &#{&}--white {
    color: #ffffff;
  }

  &#{&}:hover {
    color: $brand;
  }
}
</style>
