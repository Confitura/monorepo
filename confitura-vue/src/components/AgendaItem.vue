<template>
    <div class="agendaItem">
        <div v-if="entry.roomLabel" class="agendaItem__room">{{entry.roomLabel}}</div>
        <div v-if="entry.presentationId">
            <div class="agendaItem__title">{{entry.presentation.title}}</div>
            <div class="agendaItem__speakers">
                <span class="agendaItem__speaker" v-for="speaker in entry.speaker">{{speaker.name}}</span>
            </div>
        </div>
        <div v-else class="agendaItem__label">
            {{entry.label}}
        </div>
    </div>
</template>

<script lang="ts">
  import { Component, Prop, Vue } from 'vue-property-decorator';
  import { AgendaEntry } from '@/views/Agenda.vue';

  @Component({})
  export default class AgendaItem extends Vue {
    @Prop({ required: true })
    public entry!: AgendaEntry;


  }
</script>

<style scoped lang="scss">
    @import "../assets/sizes";
    @import "../assets/media";
    @import "../assets/colors";

    .agendaItem {
        font-size: 1rem;
        padding: .7rem;
        display: flex;
        flex-direction: column;

        @include md() {
            padding: 1.5rem;
        }

        &__room {
            color: $brand;
            font-size: .8rem;
            margin-bottom: .5rem;
            @include md() {
                display: none;
            }
        }

        &__entry--all, &__label {
            color: $brand;
            font-weight: bold;
            font-size: 1.2rem;
            text-transform: capitalize;
        }

        &__label {
            justify-self: center;
        }

    }

    .agendaItem__speakers {
        font-size: 1rem;
        color: #767676;
        margin-top: 1rem;
    }

    .agendaItem__speaker {

        &:not(:last-child):after {
            content: ', ';
        }
    }
</style>
