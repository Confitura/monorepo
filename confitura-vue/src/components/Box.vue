<template>
    <section class="box">
        <div class="box__container">
            <slot/>
        </div>
    </section>
</template>

<script lang="ts">
  import { Component, Prop, Vue } from 'vue-property-decorator';
  import { CHANGE_HEADER_THEME } from '@/types';

  @Component({})
  export default class Box extends Vue {
    @Prop()
    public color?: string;
    private threshold: number[] = [];

    constructor() {
      super();
      for (let i = 0; i <= 1; i += 0.01) {
        this.threshold.push(i);
      }
    }

    public mounted(): void {
      const options = {
        threshold: this.threshold,
      };

      const callback: IntersectionObserverCallback = (entries) => {
        const entry = entries[0];
        const height = this.$store.state.headerHeight;
        if (entry.isIntersecting) {
          if (entry.boundingClientRect.top <= height && entry.boundingClientRect.bottom > height) {
            this.$store.commit(CHANGE_HEADER_THEME, { color: this.color });
          }
        }
      };
      const observer = new IntersectionObserver(callback, options);
      observer.observe(this.$el);
    }

  }
</script>

<style scoped lang="scss">
    @import "../assets/sizes";
    .box {
        min-height: 100vh;
        width: 100%;
        display: flex;
        flex-direction: column;

        &__container {
            max-width: 1440px;
            width: 100%;
            margin: auto;
            padding: $standard-padding $standard-padding $standard-padding*3 $standard-padding;
            box-sizing: border-box;
            text-align: left;
        }
    }
</style>