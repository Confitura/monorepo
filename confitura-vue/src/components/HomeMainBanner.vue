<template>
  <section class="banner" id="home">
    <HomeMainBannerStars />
    <div class="container">
      <div class="info">
        <div class="slogan">
          <div>It is not rocket science!</div>
        </div>
        <div class="info__live">
          <div class="info__live-text">Thank you!</div>
        </div>
        <!--        <TheTimer />-->
        <div class="time-and-place">
          <div>{{ date }},&nbsp;</div>
          <div class="place">
            Warsaw Expo XXI
            <a
              class="place__link"
              href="https://www.google.com/maps/place/Warszawskie+Centrum+EXPO+XXI/@52.224893,20.9599152,17z/data=!3m1!4b1!4m5!3m4!1s0x471ecb60b93336e5:0xf7d1cb012bb9954d!8m2!3d52.224893!4d20.962104"
              target="_blank"
              rel="noopener"
            >
              <i class="fas fa-map-marker-alt"></i>
            </a>
          </div>
        </div>
      </div>
      <HomeMainBannerIllustration />
    </div>
    <img class="rocket-icon" src="../assets/rocket.svg" alt="rocket" />
  </section>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import TheTimer from "@/components/TheTimer.vue";
import HomeMainBannerIllustration from "@/components/HomeMainBannerIllustration.vue";
import { CHANGE_HEADER_THEME } from "@/types";
import dayjs from "dayjs";
import HomeMainBannerStars from "@/components/HomeMainBannerStars.vue";

@Component({
  components: { HomeMainBannerStars, HomeMainBannerIllustration, TheTimer }
})
export default class HomeMainBanner extends Vue {
  public date: string = dayjs(this.$store.state.date).format("DD.MM.YYYY");
  private threshold: number[] = [];
  private observer: IntersectionObserver | null = null;

  constructor() {
    super();
    for (let i = 0; i <= 1; i += 0.01) {
      this.threshold.push(i);
    }
  }

  public mounted(): void {
    const options = {
      threshold: this.threshold
    };

    const callback: IntersectionObserverCallback = entries => {
      const [entry] = entries;
      if (entry.isIntersecting) {
        if (entry.boundingClientRect.top < -100) {
          this.$store.commit(CHANGE_HEADER_THEME, { color: "white" });
        } else {
          this.$store.commit(CHANGE_HEADER_THEME, { color: "default" });
        }
      }
    };
    this.observer = new IntersectionObserver(callback, options);
    this.observer.observe(this.$el);
  }

  protected beforeDestroy() {
    if (this.observer !== null) {
      this.observer.disconnect();
    }
  }
}
</script>

<style scoped lang="scss">
@import "../assets/fonts";
@import "../assets/sizes";
@import "../assets/media";
@import "../assets/colors";

.banner {
  box-sizing: border-box;
  min-height: calc(100vh - 15vh);
  padding-top: 15vh;
  overflow: hidden;
  padding-left: $standard-padding;
  padding-right: $standard-padding;
  position: relative;
  @include md() {
    height: 100vh;
    padding-top: 0;
  }
}

.container {
  max-width: $max-width;
  margin: auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  align-items: center;
  z-index: 20;
  min-height: 80vh;
  position: relative;
  @include md() {
    flex-direction: row;
    min-height: unset;
  }
}

.info {
  z-index: 20;
  text-align: left;
  display: grid;
  grid-row-gap: 2rem;
  @include md() {
    margin-left: 50px;
    grid-row-gap: 4rem;
  }
}

.slogan {
  font-family: $font-bold;
  color: #ffffff;
  font-size: 2rem;
  @include xxl() {
    font-size: 3.75rem;
  }
}

.sub-slogan {
  font-family: $font-bold;
  color: #ffffff;
  font-size: 1.3rem;
  padding-top: 1rem;
}

.time-and-place {
  color: #ffffff;
  display: flex;
  flex-direction: column;
  font-size: 1.7rem;
  @include md() {
    flex-direction: row;
    font-size: 1.5rem;
  }
}

.place {
  display: flex;
  white-space: nowrap;
}

.place__link {
  margin-left: 10px;
  color: #ffffff;
}

.rocket-icon {
  display: none;
  @include md() {
    display: block;
    position: relative;
    bottom: 5rem;
    z-index: 2;
    animation: rocket-jump 2s 3;
    animation-delay: 5s;
    margin: auto;

    @keyframes rocket-jump {
      0% {
        transform: translateY(0);
      }
      50% {
        transform: translateY(-10px);
      }
      100% {
        transform: translateY(0px);
      }
    }
  }
}

.info__live {
  color: $brand;
}
.info__live-text {
  text-transform: uppercase;
  font-weight: bold;
  font-size: 2.5rem;
}
.info__live-link {
  font-size: 2rem;
}
</style>
