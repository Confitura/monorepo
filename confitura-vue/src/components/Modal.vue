<template>
  <div class="modal__container">
    <div class="modal">
      <div class="modal__header">
        <span class="closeButton" @click="close()"
          ><i class="fas fa-times"></i
        ></span>
      </div>
      <div class="modal__body">
        <slot></slot>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Emit, Vue } from "vue-property-decorator";
import PresentationBox from "@/components/PresentationBox.vue";

@Component({
  components: { PresentationBox }
})
export default class Modal extends Vue {
  @Emit()
  public close() {
    return true;
  }
}
</script>

<style scoped lang="scss">
@import "../assets/sizes";
@import "../assets/colors";

.modal__container {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  max-height: 100vh;
  z-index: 1000;
  box-sizing: border-box;
  overflow: scroll;
  background-color: rgba(0, 0, 0, 0.2);
  @include md() {
    display: flex;
    justify-content: center;
    align-items: center;
  }
}

.modal {
  background-color: #ffffff;
  min-height: 100vh;
  box-shadow: 10px 10px 20px #767676;
  display: flex;
  flex-direction: column;

  @include md() {
    width: 800px;
    min-height: 500px;
    margin: auto;
  }
}

.modal__header {
  padding-right: 0.5rem;
  padding-top: 0.5rem;
}

.modal__body {
  padding: 0 2rem 2rem;
  display: flex;
  flex-direction: column;
  flex-grow: 1;
}

.closeButton {
  cursor: pointer;
  font-size: 2rem;
  float: right;

  &:hover {
    color: $brand;
  }
}
</style>
