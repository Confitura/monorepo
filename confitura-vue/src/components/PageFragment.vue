<template>
  <div>
    <button class="edit-button" v-if="isEditable" @click="edit()">Edit</button>
    <button class="save-button" v-if="isEditing" @click="save()">Save</button>
    <button class="cancel-button" v-if="isEditing" @click="cancel()">
      Cancel
    </button>
    <section v-if="isViewing" v-html="html"></section>
    <section v-if="isEditing">
      <textarea class="editor" v-model="contentInEdit"></textarea>
    </section>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";
import axios from "axios";
import { Page } from "@/types";
import marked from "marked";

@Component({
  components: {}
})
export default class PageFragment extends Vue {
  @Prop({ required: true })
  public name!: string;

  public content: string = "";
  public contentInEdit = "";
  public editMode = false;

  public mounted() {
    axios
      .get<Page>(`/api/pages/${this.name}`)
      .then(response => response.data.content)
      .then((content: string) => (this.content = content));
  }

  public edit() {
    this.editMode = true;
    this.contentInEdit = this.content;
  }

  public cancel() {
    this.editMode = false;
  }

  public save() {
    axios
      .post<void>(`/api/pages`, { id: this.name, content: this.contentInEdit })
      .then(() => (this.content = this.contentInEdit))
      .then(() => this.cancel());
  }

  public get isEditable(): boolean {
    return this.$store.getters.isAdmin && this.isViewing;
  }

  public get isViewing() {
    return !this.editMode;
  }

  public get isEditing() {
    return this.editMode;
  }

  public get html() {
    return marked(this.content);
  }
}
</script>

<style scoped>
.editor {
  width: 100%;
  height: 100vh;
}

.edit-button,
.save-button,
.cancel-button {
  margin: 0 1rem 1rem 0;
}
</style>
