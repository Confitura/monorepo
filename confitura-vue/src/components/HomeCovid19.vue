<template>
    <Box color="white" id="covid-19">
        <div class="about-us"><h2
                class="header">Confitura 2021 canceled</h2>
            <div class="main-info">
                We regret to inform you that Confitura 2021 has been canceled. The second year in a row, the conference will not take place due to the Corona Pandemic.
                We strongly believe that after those two tough years, we will back stronger than ever!
                Hope to see you in 2022! Stay safe and take the sentimental journey through the last eight editions of Confitura. Hop on our
                <a href="https://www.youtube.com/user/confiturapl"
                   target="_blank">
                    youtube channel
                </a>
                .
            </div>
        </div>
    </Box>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import Box from "@/components/Box.vue";

@Component({
  components: { Box }
})
export default class HomeCovid19 extends Vue {}
</script>

<style scoped lang="scss">
@import "../assets/colors";
@import "../assets/fonts";
@import "../assets/media";

.about-us {
  display: grid;
  grid-row-gap: 2rem;
  grid-template-areas: "h " "mi";
  @include lg() {
    grid-column-gap: 8rem;
    grid-row-gap: 52px;
    align-content: center;
    grid-template-columns: 1fr
  }
}

.header {
  grid-area: h;
  font-family: $font-bold;
  font-size: 3rem;
  color: $brand;
}

.main-info {
  grid-area: mi;
  font-size: 1.75rem;
}

.additional-info {
  grid-area: ai;
  font-family: $font-bold;
  font-size: 1.75rem;
  color: $brand;
}
</style>
