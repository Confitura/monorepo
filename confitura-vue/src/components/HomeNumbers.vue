<template>
  <Box class="numbers" color="red" id="numbers">
    <div class="numbers-container">
      <div class="info">
        <h1 class="header">
          <span>confitura'18</span>
          <span>in numbers</span>
        </h1>
        <div class="subinfo">
          Everything has started in 2007, when the group of passionate Java
          developers decided to create a programming event for the community.
          Today we're running the conference for more than 1800 attendees. 35
          talks in 5 parallel sessions give an opportunity to learn new things
          to everyone! Our live streams and recordings of talks gain enormous
          popularity on the web. That's why organizing this event gives
          satisfaction to more than 30 people from Confitura's team. That eagers
          us to organize another Confitura, this year it will be the 13th.
        </div>
      </div>
      <HomeNumbersPictogram
        :src="require('../assets/speakers.svg')"
        value="38"
        unit="speakers"
      />
      <HomeNumbersPictogram
        :src="require('../assets/sessions.svg')"
        value="5"
        unit="tracks"
      />
      <HomeNumbersPictogram
        :src="require('../assets/presentations.svg')"
        value="35"
        unit="presentations"
      />
      <HomeNumbersPictogram
        :src="require('../assets/views.svg')"
        value="693k"
        unit="youtube channel views"
      />
      <HomeNumbersPictogram
        :src="require('../assets/videos.svg')"
        value="241"
        unit="youtube videos"
      />
      <HomeNumbersPictogram
        :src="require('../assets/participants.svg')"
        value=">1800"
        unit="participants"
      />
    </div>
  </Box>
</template>

<script lang="ts">
import HomeNumbersPictogram from "./HomeNumbersPictogram.vue";
import Box from "./Box.vue";

export default {
  name: "HomeNumbers",
  components: { Box, HomeNumbersPictogram }
};
</script>

<style scoped lang="scss">
@import "../assets/colors";
@import "../assets/fonts";
@import "../assets/media";

.numbers {
  background-color: $brand !important;
}

.numbers-container {
  display: grid;
  align-content: center;
  color: #ffffff;
  grid-row-gap: 4rem;
  @include lg() {
    grid-template-columns: 2fr 1fr 1fr 1fr;
    grid-row-gap: 3rem;
  }
}

.info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: left;
  @include lg() {
    max-width: 450px;
    grid-row: 1 / span 2;
  }
}

.header {
  font-family: $font-bold;
  font-size: 3rem;
}

.subinfo {
  font-size: 1.75rem;
}
</style>
