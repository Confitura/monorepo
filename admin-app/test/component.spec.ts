import { describe, expect, it } from 'vitest'
import DialogConfirm from '@/components/DialogConfirm.vue'
import { mountWithVuetify } from './helpers'
import { VCard } from 'vuetify/components'

describe('component DialogConfirm.vue', () => {
  it('should not open', async () => {
    const wrapper = mountWithVuetify(DialogConfirm)
    expect(wrapper.text()).toBe('')
  })

  it('should open and close', async () => {
    const wrapper = mountWithVuetify(DialogConfirm)
    wrapper.vm.open('message')
    await nextTick()
    const cardWrapper = wrapper.getComponent(VCard)
    expect(cardWrapper.text()).toContain('message')
    expect(cardWrapper.html()).toMatchSnapshot()
    expect(cardWrapper.find('button').exists()).toBe(true)

    await cardWrapper.find('button').trigger('click')
    await nextTick()
    expect(cardWrapper.isVisible()).toBe(false)
  })
})
