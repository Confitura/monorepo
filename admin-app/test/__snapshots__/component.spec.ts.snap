// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component DialogConfirm.vue > should open and close 1`] = `
"<div class="v-card v-theme--light v-card--density-default v-card--variant-elevated" style="z-index: -1;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class="v-card-text font-weight-bold d-flex"><i class="mdi-alert-circle mdi v-icon notranslate v-theme--light v-icon--size-default text-warning mr-2" aria-hidden="true"></i><span>message</span></div>
  <div class="v-card-actions">
    <div class="v-spacer"></div><button type="button" class="v-btn v-btn--slim v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Cancel</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-btn--slim v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Confirm</span>
      <!---->
      <!---->
    </button>
    <div class="v-spacer"></div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
